Mathlib/RingTheory/Flat/Basic.lean

```diff
@@ -269,6 +271,86 @@ lemma iff_lTensor_preserves_injective_linearMap [Small.{v} R] :
       (L : N →ₗ[R] N'), Function.Injective L → Function.Injective (L.lTensor M) := by
   simp_rw [iff_rTensor_preserves_injective_linearMap, LinearMap.lTensor_inj_iff_rTensor_inj]
 
+variable (R M) in
+lemma lTensor_exact [Small.{v} R] [flat : Flat R M] ⦃N N' N'' : Type v⦄
```

Comment:
I find it strange that these universe constraints exists. I thought the main purpose of the nonstandard definitions is to avoid these.

Reply:
This universe constraint is harder to avoid, it comes from Baer's criterion where we need to lift every ideal to a module. So if module lives in a universe that is smaller than the ring, Baer's criterion might not be true. @alreadydone has thought a lot about this.

Reply:
I think the only explicit variables should be `exact` and `M`. Same in `rTensor_exact`.